src/DroneSafetyApi/Models/Heatmap.cs

```diff
@@ -1,12 +1,81 @@
-﻿using System;
+﻿using Newtonsoft.Json;
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Text;
 using System.Threading.Tasks;
 
 namespace DroneSafetyApi.Models
 {
-    public class Heatmap
+
+    public class HeatMap
     {
-        public int[][] Values { get; set; }
+        private int[,] mHeatMap;
+        private double mStartX;
+        private double mStartY;
+        private int mDecimalPlaces;
+        private double mResolution;
+
+        public double GetResolution()
+        {
+            return mResolution;
+        }
+        public HeatMap(double minX, double maxX, double minY, double maxY, int decimalPlaces)
+        {
+            double rangeX = maxX - minX;
+            double rangeY = maxY - minY;
+
+            mStartX = minX;
+            mStartY = minY;
+
+            mResolution = (double)(1 / Math.Pow(10, mDecimalPlaces));
+
+            int width = (int)(rangeX * Math.Pow(10, mDecimalPlaces));
+            int height = (int)(rangeY * Math.Pow(10, mDecimalPlaces));
+
+            mHeatMap = new int[width, height];
+            mDecimalPlaces = decimalPlaces;
+        }
+
+        private double[] indexToGPS(int x, int y)
+        {
+            double lat = (double)Math.Round(mStartX + mResolution * x, mDecimalPlaces);
+            double lon = (double)Math.Round(mStartY + mResolution * y, mDecimalPlaces);
+            return new double[] { lat, lon };
+        }
+
+        private int[] GPSToIndex(double x, double y)
+        {
+            x = (double)Math.Round(x, mDecimalPlaces);
+            y = (double)Math.Round(x, mDecimalPlaces);
+            return new int[] { (int)((x - mStartX) / mResolution), (int)((y - mStartY) / mResolution) };
+        }
+
+        public void AddHazard(double x, double y, int v)
+        {
+            int[] index = GPSToIndex(x, y);
+            mHeatMap[index[0], index[1]] += v;
+        }
+
+        public string GetJsonConversion()
+        {
+            var list = new List<dynamic>();
+            for (int i = 0; i < mHeatMap.GetLength(0); i++)
+                for (int j = 0; j < mHeatMap.GetLength(1); j++)
+                {
+                    double[] GPS = indexToGPS(i, j);
+                    dynamic jsonObject = new
+                    {
+                        x = GPS[0],
+                        y = GPS[1],
+                        value = mHeatMap[i, j]
+                    };
+                    list.Add(jsonObject);
+                }
+            return JsonConvert.SerializeObject(list);
+        }
+
+
+
     }
-}
+}
```

Comment:
Visual studio created the files with extra blank line so I'm sure this setting can be fixed